### PR TITLE
Add logic - when state is absent

### PR DESCRIPTION
**Pull request by mnecas**
Issue #13

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically.
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

**Comment by machacekondra** [comment]
ci add to whitelist

File: tasks/vm_state_absent.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+- block:
```
Comment (machacekondra) [inline tasks/vm_state_absent.yml]: This block seems useless, please remove it.

File: tasks/vm_state_absent.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+  - name: "Remove VM '{{ current_vm.name }}''"
+    ovirt_vms:
+      auth: "{{ ovirt_auth }}"
+      state: "absent"
+      name: "{{ current_vm.name }}"
```
Comment (machacekondra) [inline tasks/vm_state_absent.yml]: please add new line at the end

File: tasks/vm_state_absent.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+- block:
+  - name: "Remove VM '{{ current_vm.name }}''"
```
Comment (machacekondra) [inline tasks/vm_state_absent.yml]: At the end you have two `'` please remove one.

File: tasks/main.yml [hunk 4]
```diff
@@ -1,142 +1,53 @@
...
-      - "wait_for_ip"
-      - "item.profile.state | default('present') != 'stopped'"
-
-  always:
-    - name: Logout from oVirt
-      ovirt_auth:
-        state: absent
-        ovirt_auth: "{{ ovirt_auth }}"
-      when: not loggedin.skipped | default(false)
+- name: Login to oVirt
```
Comment (machacekondra) [inline tasks/main.yml]: This should be in block and at the end there should be:

```yaml
- block:
  ...
  always:
    - name: Logout from oVirt
      ovirt_auth:
        state: absent
        ovirt_auth: "{{ ovirt_auth }}"
      when: not loggedin.skipped | default(false)
```

File: tasks/main.yml [hunk 5]
```diff
@@ -1,142 +1,61 @@
...
+    username: "{{ engine_user | default(omit) }}"
+    password: "{{ engine_password | default(omit) }}"
+    ca_file: "{{ engine_cafile | default(omit) }}"
+    insecure: "{{ engine_insecure | default(true) }}"
+  when: ovirt_auth is undefined
+  register: loggedin
+
+- name: Check if VMs are correct
+  fail:
+    msg:"'vms' variable does not contain mandatory parameter '{{ item[1] }}' "
```
Comment (machacekondra) [inline tasks/main.yml]: Add space after `msg:`
Remove traling space at the end.

File: tasks/vm_state_present.yml [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    until: "ovirt_vms | first | json_query(query) | length > 0"
+    retries: "{{ vm_infra_wait_for_ip_retries }}"
+    delay: "{{ vm_infra_wait_for_ip_delay }}"
+    vars:
+      query: "reported_devices[*].ips[*].address"
+    when:
+      - "wait_for_ip"
+      - "item.profile.state | default('present') != 'stopped'"
+
+  always:
```
Comment (machacekondra) [inline tasks/vm_state_present.yml]: This should be in `main.yml`

File: tasks/vm_state_present.yml [hunk 7]
```diff
@@ -0,0 +1,124 @@
+---
+- block:
```
Comment (machacekondra) [inline tasks/vm_state_present.yml]: This block can be removed.

File: tasks/main.yml [hunk 8]
```diff
@@ -1,142 +1,61 @@
...
+- name: Delete VM check profile
+  include_tasks: vm_state_absent.yml
+  with_items: "{{ vms }}"
+  loop_control:
+    loop_var: current_vm
+  when: 
+    - current_vm.profile.state is defined 
+    - current_vm.profile.state == 'absent' 
+
+- name: Override vms list
```
Comment (machacekondra) [inline tasks/main.yml]: Define a list of all VMs not in absent state

File: tasks/vm_state_present.yml [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+  - name: Create VMs
+    include: create_vms.yml
+    with_items: "{{ vms }}"
```
Comment (machacekondra) [inline tasks/vm_state_present.yml]: You can use in this tasks file `create_vms` isntead `vms`, so you don't need to rewrite it in `main.yml`

File: tasks/main.yml [hunk 10]
```diff
@@ -1,142 +1,61 @@
...
+
+- name: Check if VMs are correct
+  fail:
+    msg:"'vms' variable does not contain mandatory parameter '{{ item[1] }}' "
+  when: item[1] not in item[0]
+  with_nested:
+    - "{{ vms }}"
+    - ['name', 'profile']
+
+- name: Split list of VMs
```
Comment (machacekondra) [inline tasks/main.yml]: Why not have this and `Split list of VMs check profile` in one task?

File: tasks/main.yml [hunk 11]
```diff
@@ -1,142 +1,61 @@
...
+  set_fact: 
+    create_vms: "{{ create_vms | default([]) + [item]}}"
+  with_items: "{{ vms }}"
+  when: 
+    - item.state is defined
+    - item.state == 'present'
```
Comment (machacekondra) [inline tasks/main.yml]: it should be != ' absent' , as there can be other states then `presetn`.

File: tasks/main.yml [hunk 12]
```diff
@@ -1,142 +1,61 @@
...
+
+- name: Split list of VMs check profile
+  set_fact: 
+    create_vms: "{{ create_vms | default([]) + [item]}}"
+  with_items: "{{ vms }}"
+  when:
+    - item.profile.state is defined 
+    - item.profile.state == 'present'
+
+- name: Delete VM
```
Comment (machacekondra) [inline tasks/main.yml]: I think this can be in one task as well.

**Pull request merged**